File: Structural/Bridge/Python/device.py
# device.py

from abc import ABC, abstractmethod

class Device(ABC):
  @abstractmethod
  def turn_on(self):
    pass

  @abstractmethod
  def turn_off(self):
    pass

  @abstractmethod
  def set_volume(self, volume: int):
    pass

  @abstractmethod
  def get_volume(self) -> int:
    pass

  @abstractmethod
  def is_on(self) -> bool:
    pass

File: Structural/Bridge/JavaScript/device.js
// device.js

class Device {
  turnOn() {
    throw new Error("Method 'turnOn()' must be implemented.");
  }

  turnOff() {
    throw new Error("Method 'turnOff()' must be implemented.");
  }

  setVolume(volume) {
    throw new Error("Method 'setVolume()' must be implemented.");
  }

  getVolume() {
    throw new Error("Method 'getVolume()' must be implemented.");
  }

  isOn() {
    throw new Error("Method 'isOn()' must be implemented.");
  }
}

module.exports = Device;
